noobaa/noobaa-operator
# Add nsfs_account_config to noobaaaccount CRD

### Explain the changes
- Adding nsfs_account_config to noobaaaccount CRD
- Adding the fields to CreateAccountParams and calling it on demand

Signed-off-by: liranmauda <liran.mauda@gmail.com>

### Gap 
1.  No webhooks yet
2. No CLI is provided yet.

both will be provided in the 2 next PRs. 

### Limitation
Removing `nsfs_account_config:` is not supported in the core, which means once we are adding it to the account CRD, we cannot remove it. validating it will be done in the webhook validation PR.

### Testing Instructions:
A. Update existing account CRD
1. Create an account with CRD using the CLI.
2. Edit the account CRD to contain `nsfs_account_config:` in its spec:
```
spec:
  allow_bucket_creation: true
  allowed_buckets:
    full_permission: true
    permission_list: []
  default_resource: noobaa-default-backing-store
  nsfs_account_config:
    gid: 777
    new_buckets_path: /
    nsfs_only: true
    uid: 776
```
3. save and see that it is updating the account.
4. Change a field in the `nsfs_account_config:` and see that it is getting updates.

B. Create account CRD that contains `nsfs_account_config:`
1. use a noobaaaccount yaml which has `nsfs_account_config:` in its spec and apply it.
2. See that the account that is being created have those field.
3. Change a field in the `nsfs_account_config:` and see that it is getting updates.
